pkg/api/dashboard_public.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+// gets public dashboard
+func (hs *HTTPServer) GetPublicDashboard(c *models.ReqContext) response.Response {
+	dash, err := hs.dashboardService.GetPublicDashboard(c.Req.Context(), web.Params(c.Req)[":uid"])
+	if err != nil {
+		return handleDashboardErr(http.StatusInternalServerError, "Failed to get public dashboard", err)
+	}
+	return response.JSON(http.StatusOK, dash)
+}
+
+// gets public dashboard configuration for dashboard
+func (hs *HTTPServer) GetPublicDashboardConfig(c *models.ReqContext) response.Response {
+	pdc, err := hs.dashboardService.GetPublicDashboardConfig(c.Req.Context(), c.OrgId, web.Params(c.Req)[":uid"])
+	if err != nil {
+		return handleDashboardErr(http.StatusInternalServerError, "Failed to get public dashboard config", err)
+	}
+	return response.JSON(http.StatusOK, pdc)
+}
+
+// sets public dashboard configuration for dashboard
+func (hs *HTTPServer) SavePublicDashboardConfig(c *models.ReqContext) response.Response {
+	pdc := &models.PublicDashboardConfig{}
+	if err := web.Bind(c.Req, pdc); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+
+	dto := dashboards.SavePublicDashboardConfigDTO{
+		OrgId:                 c.OrgId,
+		DashboardUid:          web.Params(c.Req)[":uid"],
+		PublicDashboardConfig: pdc,
```

Comment:
the PublicDashboardConfig.PublicDashboard also contains an orgId (if I understand the code & model correctly) and it could be anything here

Reply:
This is now set explicitly in the dashboard service